theseus/options.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Any, Dict
+from dataclasses import dataclass
+
+import lie.options
+import torch
+
+
+def _CHECK_DTYPE_SUPPORTED(dtype):
+    if dtype not in [torch.float32, torch.float64]:
+        raise ValueError(
+            f"Unsupported data type {dtype}. "
+            "Theseus only supports 32- and 64-bit tensors."
+        )
+
+
+@dataclass
+class _TheseusGlobalOptions:
```

Comment:
Is this separate/duplicate setup for Theseus and torchlie mostly because we don't have SO2/SE2 in the torchlie yet? It would make sense to consolidate later.

Reply:
No, mainly because these will be two separate repos at some point and I didn't want the Theseus options structure depend on the other one. Ideally I would have made this follow some common interface, but I wasn't sure where to put it that it wouldn't be an issue when moving the repo out.

Reply:
But maybe we should remove the support for doing something like `th.set_global_options({f"torchlie.so3_near_pi_eps_float32": 100.0})`, and just let users do things separately.

Reply:
What I am worried about is having two say values of eps around in the code. Could we set this up as a wrapper on Theseus side calling into torchlie similar to the functional api? Then when torchlie moves to a standalone repo users still have the possibility of changing eps for the internally imported torchlie via the theseus level function. That way there is only one true value of an eps in any script/program.

Reply:
Maybe
```python
class _TheseusGlobalOptions(_TorchLieOptions):
    ...
    se2/so2 eps # remove after se2/so2 added to torchlie
    ....
```

Reply:
I'll take a look at what kind of unification we can do that's better than what we have. But the type of subclassing in your last example is something I was trying deliberately to avoid. I don't think we want for  `theseus` configuration settings to be a super set of `torchlie` configuration settings. I think they should remain separate entities.

Reply:
Fixed in #547